app/services/listing_index_service/search/sphinx_adapter.rb

```diff
@@ -105,10 +113,15 @@ def search_with_sphinx(community_id:, search:, included_models:)
           star: true,
           with: with,
           with_all: with_all,
-          order: 'sort_date DESC'
+          order: 'sort_date DESC',
+          max_query_time: 500 # Timeout and fail after 500ms
         )
 
-        {count: models.total_entries, listings: models}
+        begin
+          success_result(models.total_entries, models, includes)
+        rescue ThinkingSphinx::SphinxError => e
+          Result::Error.new(e)
```

Comment:
This is very nice! 

I have seen two types of error thrown by Sphinx that we should catch:

1. In development, Sphinx is not running.
2. In production, Sphinx is reindexing.

Have you tested does this now catch them both? I.e. have you tested that the exception thrown from those two cases are `ThinkingSphinx::SphinxError` or subclasses of it?